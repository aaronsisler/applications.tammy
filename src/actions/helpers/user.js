export const setUser = (user) => ({
    type: 'SET_USER',
    user,
});

export const editUser = (updates) => ({
    type: 'EDIT_USER',
    updates,
});

export const clearUser = () => ({
    type: 'CLEAR_USER'
});
