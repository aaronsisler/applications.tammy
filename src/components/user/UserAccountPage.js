import React from 'react';
import UserProfileContainer from './UserProfileContainer';

export default class UserAccountPage extends React.Component {
    constructor(props) {
        super(props);
    }

    componentDidMount() {
    }

    render() {
        return (
            <div id="user_account_page">
                <UserProfileContainer />
            </div>
        );
    }
}
