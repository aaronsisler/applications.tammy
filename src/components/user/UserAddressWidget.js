import React from 'react';
import PropTypes from 'prop-types';
import InputTools from 'User/tools/inputs';

export default class UserAddressWidget extends React.Component {
    inputTools = new InputTools();

    constructor(props) {
        super(props);
        this.state = {
            addressLine1: props.user.addressLine1 ? props.user.addressLine1 : '',
            addressLine2: props.user.addressLine2 ? props.user.addressLine2 : '',
            city: props.user.city ? props.user.city : '',
            state: props.user.state ? props.user.state : '',
            postalCode: props.user.postalCode ? props.user.postalCode : '',
        };
    }

    componentDidUpdate(prevProps) {
        const { user: oldUser } = prevProps;
        const { user: newUser } = this.props;

        if (oldUser.userId !== newUser.userId) {
            return this.setState(() => ({
                addressLine1: newUser.addressLine1 ? newUser.addressLine1 : '',
                addressLine2: newUser.addressLine2 ? newUser.addressLine2 : '',
                city: newUser.city ? newUser.city : '',
                state: newUser.state ? newUser.state : '',
                postalCode: newUser.postalCode ? newUser.postalCode : '',
            }));
        }
    }

    handleInputChange = (e) => {
        const inputName = e.target.name;
        const inputValue = e.target.value;

        switch (inputName) {
            case 'postalCode':
                this.handlePostalCodeValidation(inputValue);
                return;
            case 'state':
                this.handleStateAbbreviationValidation(inputValue);
                return;
            default:
                this.handleAddInput(inputName, inputValue);
                return;
        }
    }

    handlePostalCodeValidation = (inputValue) => {
        if (!inputValue || inputValue.match(/^\d{1,5}?$/)) {
            this.setState(() => ({ postalCode: inputValue }));
        }
    }

    handleStateAbbreviationValidation = (inputValue) => {
        if (!inputValue || inputValue.match(/^[A-Za-z]{1,2}$/)) {
            this.setState(() => ({ state: inputValue.toUpperCase() }));
        }
    }

    handleAddInput = (inputName, inputValue) => (
        this.setState(() => ({
            [inputName]: inputValue
        }))
    )

    handleSubmit = () => {
        const submitObject = Object.assign({}, this.state);
        this.props.onSubmit(submitObject);
    }

    render() {
        return (
            <div className="user_address_widget">
                <div className="user_address_content">
                    <div className="user_address_widget__title">
                        Address
                    </div>
                    <div className="user_address_widget__inputs">
                        <input
                            readOnly={this.props.isReadOnly}
                            type="text"
                            name="addressLine1"
                            placeholder="Address Line 1"
                            value={this.state.addressLine1}
                            onChange={this.handleInputChange}
                            onBlur={this.inputTools.handleRequiredValidation}
                        />
                        <input
                            readOnly={this.props.isReadOnly}
                            type="text"
                            name="addressLine2"
                            placeholder="Address Line 2"
                            value={this.state.addressLine2}
                            onChange={this.handleInputChange}
                        />
                        <div className="user_address_widget__inputs_location">
                            <input
                                readOnly={this.props.isReadOnly}
                                type="text"
                                name="city"
                                placeholder="City"
                                value={this.state.city}
                                onChange={this.handleInputChange}
                                onBlur={this.inputTools.handleRequiredValidation}
                            />
                            <input
                                readOnly={this.props.isReadOnly}
                                type="text"
                                name="state"
                                placeholder="State"
                                value={this.state.state}
                                onChange={this.handleInputChange}
                                onBlur={this.inputTools.handleRequiredValidation}
                            />
                            <input
                                readOnly={this.props.isReadOnly}
                                type="text"
                                name="postalCode"
                                placeholder="Postal Code"
                                value={this.state.postalCode}
                                onChange={this.handleInputChange}
                                onBlur={this.inputTools.handleRequiredValidation}
                            />
                        </div>
                    </div>
                </div>
                {!this.props.isReadOnly &&
                    <div className="user_address_widget__button">
                        <button
                            disabled={!this.state.addressLine1 || !this.state.city || !this.state.state || !this.state.postalCode}
                            onClick={this.handleSubmit}
                        >
                            Update Address
                       </button>
                    </div>
                }
            </div>
        )
    }
}

UserAddressWidget.propTypes = {
    isReadOnly: PropTypes.bool,
    user: PropTypes.object.isRequired,
    onSubmit: PropTypes.func,
};
