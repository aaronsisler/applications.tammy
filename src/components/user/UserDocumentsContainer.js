import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import DocumentUploadWidget from '../shared/document/DocumentUploadWidget';
import DocumentList from '../shared/document/DocumentList';
import LoadingPage from '../core/LoadingPage';

export class UserDocumentsContainer extends React.Component {
    constructor(props) {
        super(props);
    }

    componentDidMount() {
    }

    render() {
        return (
            <div>
                {!this.props.documents && <LoadingPage />}
                {this.props.documents &&
                    <div id="user_documents_container">
                        <DocumentUploadWidget />
                        <DocumentList
                            documents={this.props.documents}
                        />
                    </div>
                }
            </div>
        );
    }
}

const mapStateToProps = (state) => ({
    documents: state.userDocuments,
})

export default connect(mapStateToProps)(UserDocumentsContainer);

UserDocumentsContainer.propTypes = {
    documents: PropTypes.array.isRequired,
};
