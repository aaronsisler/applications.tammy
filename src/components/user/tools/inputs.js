import { requiredInputFieldClassName } from '../../../tools/constants';

export const handleRequiredValidation = (e) => {
    const inputName = e.target.name;
    const inputValue = e.target.value;

    if (!inputValue) {
        document.getElementsByName(inputName)[0].classList.add(requiredInputFieldClassName);
    }
    else {
        document.getElementsByName(inputName)[0].classList.remove(requiredInputFieldClassName);
    }
}
