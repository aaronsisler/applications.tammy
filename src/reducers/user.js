const userReducerDefaultState = null;

const userReducer = (state = userReducerDefaultState, action) => {
    switch (action.type) {
        case 'SET_USER':
            return action.user;
        case 'CLEAR_USER':
            return null;
        case 'EDIT_USER':
            return {
                ...state,
                ...action.updates
            }
        default:
            return state;
    }
};

export default userReducer;

