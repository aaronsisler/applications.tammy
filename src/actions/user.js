import database from 'Firebase/firebase';
import { setUser, editUser, clearUser } from './helpers/user';

export const startSetUser = () => (dispatch, getState) => {
    const { uid: userId } = getState().auth;

    return database.ref(`users/${userId}`).once('value').then((snapshot) => {
        dispatch(setUser(
            {
                userId: snapshot.key,
                ...snapshot.val()
            }));
    })
};

export const startEditUser = (updates) => (dispatch, getState) => {
    const { uid: userId } = getState().auth;

    return database.ref(`users/${userId}`).update(updates).then(() => {
        dispatch(editUser(updates));
    });
};

export const startClearUser = () => (dispatch) => dispatch(clearUser());
