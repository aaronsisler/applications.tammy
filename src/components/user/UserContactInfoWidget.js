import React from 'react';
import PropTypes from 'prop-types';
import { requiredInputFieldClassName } from '../../tools/constants';
import { isEmailValid } from '../../tools/email';

export default class UserNameWidget extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            isEmailValid: props.user.email ? true : false,
            email: props.user.email ? props.user.email : '',
            phoneAreaCode: props.user.phoneAreaCode ? props.user.phoneAreaCode : '',
            phonePrefix: props.user.phonePrefix ? props.user.phonePrefix : '',
            phoneLineNumber: props.user.phoneLineNumber ? props.user.phoneLineNumber : '',
            phoneExt: props.user.phoneExt ? props.user.phoneExt : '',
        };
    }

    handleEmailInput = (e) => {
        const inputValue = e.target.value;

        this.setState({ email: inputValue });
    }

    handleEmailValidation = (e) => {
        const inputName = e.target.name;
        const inputValue = e.target.value;

        if (isEmailValid(inputValue)) {
            document.getElementsByName(inputName)[0].classList.remove(requiredInputFieldClassName);
            return this.setState({ isEmailValid: true });
        }
        else {
            document.getElementsByName(inputName)[0].classList.add(requiredInputFieldClassName);
            return this.setState({ isEmailValid: false });
        }
    }

    handlePhoneNumberValidation = (e) => {
        const inputName = e.target.name;
        const inputValue = e.target.value;

        if (!inputValue) {
            return this.setState({ [inputName]: inputValue });
        }

        switch (inputName) {
            case 'phoneAreaCode':
            case 'phonePrefix':
                if (inputValue.match(/^\d{1,3}?$/)) {
                    this.setState({ [inputName]: inputValue });
                }
                return;
            case 'phoneLineNumber':
                if (inputValue.match(/^\d{1,4}?$/)) {
                    this.setState({ [inputName]: inputValue });
                }
                return;
            default:
                if (inputValue.match(/^\d{1,5}?$/)) {
                    this.setState({ [inputName]: inputValue });
                }
                return;
        }
    }

    handleSubmit = () => {
        const submitObject = Object.assign({}, this.state);
        this.props.onSubmit(submitObject);
    }

    render() {
        return (
            <div id="user_contact_info_widget">
                <div className="user_email_content" >
                    <div className="user_contact_info_content_title">
                        Email
                    </div>
                    <div className="user_email_input">
                        <input
                            readOnly={this.props.isReadOnly}
                            type="text"
                            name="email"
                            placeholder="Email"
                            className="text_input"
                            value={this.state.email}
                            onChange={this.handleEmailInput}
                            onBlur={this.handleEmailValidation}
                        />
                    </div>
                </div>
                <div className="user_phone_number_content">
                    <div className="user_contact_info_content_title">
                        Phone Number
                    </div>
                    <div className="user_phone_number_inputs">
                        <div className="user_phone_number_input">
                            <input
                                readOnly={this.props.isReadOnly}
                                type="text"
                                name="phoneAreaCode"
                                placeholder="(919)"
                                className="text_input"
                                value={this.state.phoneAreaCode}
                                onChange={this.handlePhoneNumberValidation}
                            />
                        </div>
                        <div className="user_phone_number_input">
                            <input
                                readOnly={this.props.isReadOnly}
                                type="text"
                                name="phonePrefix"
                                placeholder="123"
                                className="text_input"
                                value={this.state.phonePrefix}
                                onChange={this.handlePhoneNumberValidation}
                            />
                        </div>
                        <div className="user_phone_number_input">
                            <input
                                readOnly={this.props.isReadOnly}
                                type="text"
                                name="phoneLineNumber"
                                placeholder="4567"
                                className="text_input"
                                value={this.state.phoneLineNumber}
                                onChange={this.handlePhoneNumberValidation}
                            />
                        </div>
                        <div className="user_phone_number_input">
                            <input
                                readOnly={this.props.isReadOnly}
                                type="text"
                                name="phoneExt"
                                placeholder="Ext."
                                className="text_input"
                                value={this.state.phoneExt}
                                onChange={this.handlePhoneNumberValidation}
                            />
                        </div>
                    </div>
                </div>
                <div className="user_contact_info_widget_button">
                    {!this.props.isReadOnly &&
                        <button
                            disabled={!this.state.isEmailValid}
                            onClick={this.handleSubmit}
                            className="button"
                        >
                            Update Contact Info
                    </button>
                    }
                </div>
            </div>
        )
    }
}

UserNameWidget.propTypes = {
    isReadOnly: PropTypes.bool,
    user: PropTypes.object.isRequired,
    onSubmit: PropTypes.func.isRequired,
};
