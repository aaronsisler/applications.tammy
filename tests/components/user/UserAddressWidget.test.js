
import React from 'react';
import { shallow } from 'enzyme';
import UserAddressWidget from '../../../src/components/user/UserAddressWidget';
import user from '../../fixtures/user';
import * as inputTools from '../../../src/components/user/tools/inputs';
jest.mock('../../../src/components/user/tools/inputs', () => ({ handleRequiredValidation: jest.fn() }));

describe('UserAddressWidget', () => {
    const onSubmit = jest.fn();
    let wrapper;

    describe('when isReadOnly is FALSE', () => {
        beforeEach(() => {
            wrapper = shallow(
                <UserAddressWidget
                    isReadOnly={false}
                    onSubmit={onSubmit}
                    user={user}
                />);
        });

        afterEach(() => {
            jest.restoreAllMocks();
        });

        xit('should render UserAddressWidget correctly', () => {
            expect(wrapper).toMatchSnapshot();
        });

        describe('Address Line 1 Input', () => {
            const name = 'addressLine1';

            it('should set address line 1 on input change', () => {
                const value = '123 Evergreen Terrace';
                const target = { name, value };

                wrapper.find(`#${name}`).simulate('change', { target });

                expect(wrapper.state(name)).toBe(value);
            });

            it('should call handleRequiredValidation on blur', () => {
                wrapper.find(`#${name}`).simulate('blur');

                expect(inputTools.handleRequiredValidation).toHaveBeenCalled();
            });
        })

        it('should set address line 2 on input change', () => {
            const name = 'addressLine2';
            const value = 'Apt 456';

            wrapper.find(`#${name}`).simulate('change', {
                target: { name, value }
            });

            expect(wrapper.state(name)).toBe(value);
        });

        describe('City Input', () => {
            const name = 'city';
            it('should set city on input change', () => {
                const value = 'Durham';

                wrapper.find(`#${name}`).simulate('change', {
                    target: { value }
                });

                expect(wrapper.state(name)).toBe(value);
            });

            it('should call handleRequiredValidation on blur', () => {
                jest.spyOn(inputTools, 'handleRequiredValidation');

                wrapper.find(`#${name}`).simulate('blur');

                expect(inputTools.handleRequiredValidation).toHaveBeenCalled();
            });
        });

        describe('State Input', () => {
            const name = 'state';

            it('should set state on valid input change', () => {
                const value = 'NC';

                wrapper.find(`#${name}`).simulate('change', {
                    target: { value }
                });

                expect(wrapper.state(name)).toBe(value);
            });

            it('should NOT set state on invalid input change', () => {
                const value = '1';

                wrapper.find(`#${name}`).simulate('change', {
                    target: { value }
                });

                expect(wrapper.state(name)).toBe(user[name]);
            });

            it('should call handleRequiredValidation on blur', () => {
                jest.spyOn(inputTools, 'handleRequiredValidation');

                wrapper.find(`#${name}`).simulate('blur');

                expect(inputTools.handleRequiredValidation).toHaveBeenCalled();
            });
        });

        describe('Postal Code Input', () => {
            const name = 'postalCode';

            it('should set postal code on valid input change', () => {
                const value = '12345';

                wrapper.find(`#${name}`).simulate('change', {
                    target: { value }
                });

                expect(wrapper.state(name)).toBe(value);
            });

            it('should NOT set postal code on invalid input change', () => {
                const value = 'a';

                wrapper.find(`#${name}`).simulate('change', {
                    target: { value }
                });

                expect(wrapper.state(name)).toBe(user[name]);
            });

            it('should call handleRequiredValidation on blur', () => {
                jest.spyOn(inputTools, 'handleRequiredValidation');

                wrapper.find(`#${name}`).simulate('blur');

                expect(inputTools.handleRequiredValidation).toHaveBeenCalled();
            });
        });

        describe('Update Address button', () => {
            it('should be enabled when required fields ARE populated', () => {
                expect(wrapper.find('button').props().disabled).toBe(false);
            });

            it('should be enabled when required fields ARE populated', () => {
                wrapper.setState({
                    addressLine1: '',
                    addressLine2: '',
                    city: '',
                    state: '',
                    postalCode: ''
                });

                expect(wrapper.find('button').props().disabled).toBe(true);
            });

            it('should call onSubmit prop on button click', () => {
                const {
                    addressLine1,
                    addressLine2,
                    city,
                    state,
                    postalCode,
                } = user;

                wrapper.find('button').simulate('click');

                expect(onSubmit).toHaveBeenLastCalledWith({
                    addressLine1,
                    addressLine2,
                    city,
                    state,
                    postalCode,
                });
            });
        });
    });

    describe('when isReadOnly is TRUE', () => {
        beforeEach(() => {
            wrapper = shallow(
                <UserAddressWidget
                    isReadOnly={true}
                    onSubmit={onSubmit}
                    user={user}
                />);
        });

        xit('should render UserAddressWidget correctly', () => {
            expect(wrapper).toMatchSnapshot();
        })
    })
});
