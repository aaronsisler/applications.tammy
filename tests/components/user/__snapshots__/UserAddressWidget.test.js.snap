// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserAddressWidget when isReadOnly is FALSE should render UserAddressWidget correctly 1`] = `
<div
  id="user_address_widget"
>
  <div
    className="user_address_content"
  >
    <div
      className="user_address_content_title"
    >
      Address
    </div>
    <div
      className="user_address_inputs"
    >
      <div
        className="user_address_street_addresses"
      >
        <div
          className="user_address_input"
        >
          <input
            className="text_input"
            id="addressLine1"
            name="addressLine1"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Address Line 1"
            readOnly={false}
            type="text"
            value="123 Evergreen Terrace"
          />
        </div>
        <div
          className="user_address_input"
        >
          <input
            className="text_input"
            id="addressLine2"
            name="addressLine2"
            onChange={[Function]}
            placeholder="Address Line 2"
            readOnly={false}
            type="text"
            value="Apt. 456"
          />
        </div>
      </div>
      <div
        className="user_address_city_state_zip"
      >
        <div
          className="user_address_input"
        >
          <input
            className="text_input"
            id="city"
            name="city"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="City"
            readOnly={false}
            type="text"
            value="Durham"
          />
        </div>
        <div
          className="user_address_input"
        >
          <input
            className="text_input"
            id="state"
            name="state"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="State"
            readOnly={false}
            type="text"
            value="NC"
          />
        </div>
        <div
          className="user_address_input"
        >
          <input
            className="text_input"
            id="postalCode"
            name="postalCode"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Postal Code"
            readOnly={false}
            type="text"
            value="12345"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="user_address_widget_button"
  >
    <button
      className="button"
      disabled={false}
      onClick={[Function]}
    >
      Update Address
    </button>
  </div>
</div>
`;

exports[`UserAddressWidget when isReadOnly is TRUE should render UserAddressWidget correctly 1`] = `
<div
  id="user_address_widget"
>
  <div
    className="user_address_content"
  >
    <div
      className="user_address_content_title"
    >
      Address
    </div>
    <div
      className="user_address_inputs"
    >
      <div
        className="user_address_street_addresses"
      >
        <div
          className="user_address_input"
        >
          <input
            className="text_input"
            id="addressLine1"
            name="addressLine1"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Address Line 1"
            readOnly={true}
            type="text"
            value="123 Evergreen Terrace"
          />
        </div>
        <div
          className="user_address_input"
        >
          <input
            className="text_input"
            id="addressLine2"
            name="addressLine2"
            onChange={[Function]}
            placeholder="Address Line 2"
            readOnly={true}
            type="text"
            value="Apt. 456"
          />
        </div>
      </div>
      <div
        className="user_address_city_state_zip"
      >
        <div
          className="user_address_input"
        >
          <input
            className="text_input"
            id="city"
            name="city"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="City"
            readOnly={true}
            type="text"
            value="Durham"
          />
        </div>
        <div
          className="user_address_input"
        >
          <input
            className="text_input"
            id="state"
            name="state"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="State"
            readOnly={true}
            type="text"
            value="NC"
          />
        </div>
        <div
          className="user_address_input"
        >
          <input
            className="text_input"
            id="postalCode"
            name="postalCode"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Postal Code"
            readOnly={true}
            type="text"
            value="12345"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
